cmake/scripts/common/ArchSetup.cmake

```diff
@@ -98,11 +98,11 @@ message(STATUS "Cross-Compiling: ${CMAKE_CROSSCOMPILING}")
 message(STATUS "Execute build artefacts on host: ${CORE_HOST_IS_TARGET}")
 message(STATUS "Depends based build: ${KODI_DEPENDSBUILD}")
 
-check_type(string std::u16string HAVE_STD__U16_STRING)
-check_type(string std::u32string HAVE_STD__U32_STRING)
-check_type(string char16_t HAVE_CHAR16_T)
-check_type(string char32_t HAVE_CHAR32_T)
-check_type(stdint.h uint_least16_t HAVE_STDINT_H)
+check_type(char16_t HAVE_CHAR16_T)
+check_type(char32_t HAVE_CHAR32_T)
+check_type(std::u16string HAVE_STD__U16_STRING)
+check_type(std::u32string HAVE_STD__U32_STRING)
```

Comment:
these are guaranteed by c++14 (actually, by 11) and since we require the standard, the probes are useless no ?

Reply:
good point. iirc we had them even before c++11. I can remove them

Reply:
yes because we initially built under c++0x. you have to change some code that checks the defines as well

Reply:
actually the code using it is nowhere to be found so just removing the probes suffices.